File: fix_enc.js
const fs = require('fs');
const iconv = require('iconv-lite');

function fix() {
    const filePath = 'apps/site/views.py';
    const content = fs.readFileSync(filePath, 'utf8');
    const lines = content.split('\n');
    const newLines = [];

    for (let line of lines) {
        if (line.includes('ط') || line.includes('ظ')) {
            try {
                // The characters were mapped from UTF-8 bytes to Windows-1256.
                // We need to get the Windows-1256 bytes of this string, then decode those bytes as UTF-8.
                const buf = iconv.encode(line, 'win1256');
                const fixed = buf.toString('utf8');
                
                // if it decodes properly and doesn't contain replacement character
                if (fixed.includes('\uFFFD')) {
                    newLines.push(line);
                } else {
                    newLines.push(fixed);
                }
            } catch (e) {
                newLines.push(line);
            }
        } else {
            newLines.push(line);
        }
    }

    fs.writeFileSync(filePath, newLines.join('\n'), 'utf8');
    console.log('Fixed file.');
}

fix();

File: fix_enc.py
import os

def fix():
    with open('apps/site/views.py', 'r', encoding='utf-8') as f:
        content = f.read()

    new_content = ""
    # find lines with mojibake
    for line in content.split('\n'):
        if 'ط' in line or 'ظ' in line:
            try:
                # The text was written as utf-8, but powershell read it as windows-1252 and then saved it as utf-8.
                line = line.encode('windows-1252').decode('utf-8')
            except:
                pass
        new_content += line + '\n'

    with open('apps/site/views.py', 'w', encoding='utf-8') as f:
        f.write(new_content)

if __name__ == '__main__':
    fix()
